core/src/queue.rs

```diff
@@ -275,7 +266,7 @@ impl Queue {
                 continue;
             }
             seen.push(hash);
-            if *tx.check_signature_condition(wsv).unwrap_or(MustUse(false)) {
+            if *tx.check_signature_condition(wsv) {
```

Comment:
This check look redundant now, because we add transaction in the queue only if it's `check_signature_condition` return `true`.

The only case when `check_signature_condition` became `false` if there is another transaction in the queue which changes signature check condition for the account. 
Not sure how we should handle this case.

Reply:
Refactored by using `check_tx` matcher